lua/pac3/core/shared/movement.lua

```diff
@@ -1,52 +1,54 @@
-CreateConVar("pac_free_movement", -1, CLIENT and {FCVAR_REPLICATED} or {FCVAR_ARCHIVE, FCVAR_REPLICATED}, "allow players to modify movement. -1 apply only allow when noclip is allowed, 1 allow for all gamemodes, 0 to disable")
-
-local default = {}
-default.JumpHeight = 200
-default.StickToGround = true
-default.GroundFriction = 0.12
-default.AirFriction = 0.01
-default.Gravity = Vector(0,0,-600)
-default.Noclip = false
-default.MaxGroundSpeed = 750
-default.MaxAirSpeed = 1
-default.AllowZVelocity = false
-default.ReversePitch = false
-default.UnlockPitch = false
-default.VelocityToViewAngles = 0
-default.RollAmount = 0
-
-default.SprintSpeed = 750
-default.RunSpeed = 300
-default.WalkSpeed = 100
-default.DuckSpeed = 25
-
-default.FinEfficiency = 0
-default.FinLiftMode = "normal"
-default.FinCline = false
+local movementConvar = CreateConVar("pac_free_movement", -1, CLIENT and {FCVAR_REPLICATED} or {FCVAR_ARCHIVE, FCVAR_REPLICATED}, "allow players to modify movement. -1 apply only allow when noclip is allowed, 1 allow for all gamemodes, 0 to disable")
+
+local default = {
+	JumpHeight = 200,
+	StickToGround = true,
+	GroundFriction = 0.12,
+	AirFriction = 0.01,
+	Gravity = Vector(0,0,-600),
+	Noclip = false,
+	MaxGroundSpeed = 750,
+	MaxAirSpeed = 1,
+	AllowZVelocity = false,
+	ReversePitch = false,
+	UnlockPitch = false,
+	VelocityToViewAngles = 0,
+	RollAmount = 0,
+
+	SprintSpeed = 750,
+	RunSpeed = 300,
+	WalkSpeed = 100,
+	DuckSpeed = 25,
+
+	FinEfficiency = 0,
+	FinLiftMode = "normal",
+	FinCline = false
+}
 
 if SERVER then
 	util.AddNetworkString("pac_modify_movement")
 
 	net.Receive("pac_modify_movement", function(len, ply)
-		local cvar = GetConVarNumber("pac_free_movement")
-		if cvar == 1 or (cvar == -1 and hook.Run("PlayerNoClip", ply, true)) then
-			local str = net.ReadString()
-			if str == "disable" then
-				ply.pac_movement = nil
-			else
-				if default[str] ~= nil then
-					local val = net.ReadType()
-					if type(val) == type(default[str]) then
-						ply.pac_movement = ply.pac_movement or table.Copy(default)
-						ply.pac_movement[str] = val
-					end
+		local cvar = movementConvar:GetInt()
+		if cvar ~= 1 or cvar ~= -1 or not hook.Run("PlayerNoClip", ply, true) then return end
```

Comment:
This if statement is wrong.

Reply:
I think the correct conversion is `if cvar ~= 1 and (cvar ~= -1 or not hook.Run("PlayerNoClip", ply, true)) then`

Reply:
Yeah i was just testing this, i'll fix it

Reply:
I think you should use `hook.Run("PlayerNoClip", ply, true)~=false` instead of `not hook.Run("PlayerNoClip", ply, true)` as well.

Reply:
I think it should be `== false` because it'd return at true which we want

Reply:
Yeah after testing, `==` is what we need.

Reply:
`if cvar == 0 or (cvar == -1 and hook.Run("PlayerNoClip", ply, true)==false) then`

Reply:
Yeah that's better and easier to read, I'll use that 👍